File: endpoints/photos.py
from silence.decorators import endpoint

@endpoint(
    route="/photos",
    method="GET",
    sql="SELECT * FROM Photos ORDER BY date DESC"
)
def get_all():
    pass
############################

@endpoint(
    route="/photos/visibility/$visibility",
    method="GET",
    sql="SELECT * FROM Photos WHERE visibility=$visibility ORDER BY date DESC"
)
def get_all_visibility():
    pass
############################

@endpoint(
    route="/photos/$photoId",
    method="GET",
    sql="SELECT * FROM Photos WHERE photoId=$photoId"
)
def get_by_id():
    pass

############################

@endpoint(
    route="/photos/categories/$category",
    method="GET",
    sql="SELECT * FROM Photos WHERE category=$category ORDER BY date DESC"
)
def get_by_category():
    pass

############################

@endpoint(
    route="/photos/users/$userId",
    method="GET",
    sql="SELECT * FROM Photos WHERE userId=$userId ORDER BY date DESC"
)
def get_by_user():
    pass

############################

@endpoint(
    route="/photos/week",
    method="GET",
    sql="SELECT * FROM Photos WHERE (date BETWEEN DATE_SUB(NOW(), INTERVAL 7 DAY) AND NOW()) AND visibility='Public' ORDER BY date DESC"
)
def get_this_week():
    pass

############################
  
@endpoint(
    route="/photos",
    method="POST",
    sql="INSERT INTO Photos (userId, title, description, url, category, visibility) VALUES ($userId, $title, $description, $url, $category, $visibility)",
    description="Creates a new photo",
)
def create(userId, title, description, url, category, visibility):
    pass

###############################################################################

@endpoint(
    route="/photos/$photoId",
    method="PUT",
    sql="UPDATE Photos SET userId=$userId, title = $title, description = $description, url = $url, category = $category, visibility = $visibility WHERE photoId = $photoId",
    description="Updates an existing photo",
)
def update(userId, title, description, url, category, visibility):
    pass

###############################################################################

@endpoint(
    route="/photos/$photoId",
    method="DELETE",
    sql="DELETE FROM Photos WHERE photoId = $photoId",
    description="Removes a photo",
)
def delete():
    pass
    

File: docs/js/api/photos.js
"use strict";

import { BASE_URL, requestOptions } from "./common.js";

const photosAPI = {
    getAll: function () {
        return new Promise(
            function (resolve, reject) {
                axios.get(`${BASE_URL}/photos`, requestOptions)
                    .then(response => resolve(response.data))
                    .catch(error => reject(error.response.data.message));
            }
        );
    },
    getByVisibility: function (visibility) {
        return new Promise(
            function (resolve, reject) {
                axios.get(`${BASE_URL}/photos/visibility/${visibility}`, requestOptions)
                    .then(response => resolve(response.data))
                    .catch(error => reject(error.response.data.message));
            }
        );
    },
    getById: function (photoId) {
        return new Promise(
            function (resolve, reject) {
                axios.get(`${BASE_URL}/photos/${photoId}`, requestOptions)
                    .then(response => resolve(response.data))
                    .catch(error => reject(error.response.data.message));
            }
        );
    },
    getByCategory: function (categoryName) {
        return new Promise(
            function (resolve, reject) {
                axios.get(`${BASE_URL}/photos/categories/${categoryName}`, requestOptions)
                    .then(response => resolve(response.data))
                    .catch(error => reject(error.response.data.message));
            }
        );
    },
    getByUser: function (userId) {
        return new Promise(
            function (resolve, reject) {
                axios.get(`${BASE_URL}/photos/users/${userId}`, requestOptions)
                    .then(response => resolve(response.data))
                    .catch(error => reject(error.response.data.message));
            }
        );
    },
    getThisWeek: function () {
        return new Promise(
            function (resolve, reject) {
                axios.get(`${BASE_URL}/photos/week`, requestOptions)
                    .then(response => resolve(response.data))
                    .catch(error => reject(error.response.data.message));
            }
        );
    },
    create: function (formData) {
        return new Promise(function (resolve, reject) {
            axios
                .post(`${BASE_URL}/photos`, formData, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },
    update: function (photoId, formData) {
        return new Promise(function (resolve, reject) {
            axios
                .put(`${BASE_URL}/photos/${photoId}`, formData, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });
    },

    delete: function (photoId) {
        return new Promise(function (resolve, reject) {
            axios
                .delete(`${ BASE_URL }/photos/${photoId}`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        });

    }
};
export { photosAPI }